lib/bridge.js

```diff
@@ -1074,7 +1056,7 @@ BridgeContext.prototype.get = function(roomStore, userStore) {
         if (mxSender) {
             self.senders.matrix = mxSender;
         }
-    });
+    }).catch(e => {throw wrap(e, Error, "Could not retrieve bridge context");});
```

Comment:
My advice for this feature is to disable it by default and stick a config flag in the constructor at the top like we do for things like room upgrades. It means bridges that rely on previous behaviour will not explode, but it will still allow them to opt in with a one liner :)

Reply:
I am not sure which feature you are talking about here. With this line an Error is thrown, without this line an Error is thrown :thinking: Could you please elaborate a bit more?

Reply:
Yep, sorry. It was tricky to find the best line to use for context.

Automatically sending signal events into the room upon failure is probably not something we want on by default (which is what I assume this line is doing?). The reason for this is that it might do unexpected and unwanted things for users who are just upgrading their bridge version. Could we put automatic signaling behind a constructor flag?

Reply:
Oh, I get it, you are talking about this whole feature :D

https://github.com/matrix-org/matrix-appservice-bridge/blob/dcd323e17a6d0314f09756c21f0a90232b538601/lib/bridge.js#L794

This is your feature switch, essentially. Only if the bridge rejects with an `EventNotHandledError` or one of its subtypes, the behavior is triggered. By that, bridges which do not change anything, don't get anything new.

The specific line referenced by you is pretty close to a no-op: In the case of an Error already thrown, it catches it and rethrows a new one, just with another error message.

Reply:
That's very clever and I totally misunderstood it. Makes more sense to me now :)